libvcx/src/aries/handlers/connection/connection.rs

```diff
@@ -676,6 +684,23 @@ Get messages received from connection counterparty.
     }
 
     pub fn to_string(&self) -> String {
+        serde_json::to_string(&self).unwrap()
+    }
+
+    pub fn from_string(connection_data: &str) -> VcxResult<Connection> {
+        let conn = serde_json::from_str(connection_data)
+            .map_err(|err| VcxError::from_msg(VcxErrorKind::InvalidJson, format!("Cannot deserialize Connection: {:?}", err)))?;
+        Ok(conn)
+    }
```

Comment:
I guess these can stay - at least the client needs no direct dependency on serde_json. In that case, please remove the `unwrap` and the `Ok(conn)` line should not be necessary.

Reply:
In this case I disagree enclosing two linked functions calls, on 2 lines, with error propagation, into `Ok`.

Reply:
unwrap removed

Reply:
Nobody is suggesting that.
```
    pub fn from_string(connection_data: &str) -> VcxResult<Connection> {
        serde_json::from_str(connection_data)
            .map_err(|err| VcxError::from_msg(VcxErrorKind::InvalidJson, format!("Cannot deserialize Connection: {:?}", err)))
    }
```